### ISSUE TITLE
different prediction result

### ISSUE THREAD

**Issue by zdarktknight** [aaxwaz/Multivariate-Time-Series-forecast-using-seq2seq-in-TensorFlow#2]
It is a very good example.
I can almost reproduce every figure, but the prediction is really bad. I have a plot looks like this. 
![figure_1](https://user-images.githubusercontent.com/8602335/32984666-d5d78256-cc78-11e7-9418-1c1487c55949.png)
I am using Tensorflow 0.14

**Comment by aaxwaz** [comment]
Hard for me to guess out the reason simply based on your plotted graph. Did you follow exactly my steps? 
On the other hand, I am using TF 1.2 and above, maybe you could try the latest version?

**Comment by zdarktknight** [comment]
Thank you for your quick reply.
I am using TF 1.4
I followed exactly your step using your `build_model_basic.py` code.
I removed following lines from **line [269]** 

           temp_saver = rnn_model['saver']()
           save_path = temp_saver.save(sess, os.path.join('./', 'univariate_ts_model0'))
    print("Checkpoint` saved at: ", save_path)print("Checkpoint saved at: ", save_path) 
and **line [271]**
`saver = rnn_model['saver']().restore(sess, os.path.join('./', 'univariate_ts_model0'))`

The figure looks like a tendency prediction. Is it because of the `build_model_basic.py` I use? But I used the same file as yours.

Here is my code following your step with above lines removed: 
	
	from build_model_basic import * 

	## Parameters
	learning_rate = 0.01
	lambda_l2_reg = 0.003  

	## Network Parameters
	# length of input signals
	input_seq_len = 15 
	# length of output signals
	output_seq_len = 20 
	# size of LSTM Cell
	hidden_dim = 64 
	# num of input signals
	input_dim = 1
	# num of output signals
	output_dim = 1
	# num of stacked lstm layers 
	num_stacked_layers = 2 
	# gradient clipping - to avoid gradient exploding
	GRADIENT_CLIPPING = 2.5


	total_iteractions = 100
	batch_size = 16
	KEEP_RATE = 0.5
	train_losses = []
	val_losses = []

	x = np.linspace(0, 30, 105)
	train_data_x = x[:85]

	rnn_model = build_graph(feed_previous=False)

	saver = tf.train.Saver()

	init = tf.global_variables_initializer()
	with tf.Session() as sess:

	    sess.run(init)
	    
	    for i in range(total_iteractions):
	        batch_input, batch_output = generate_train_samples(batch_size=batch_size)
	        
	        feed_dict = {rnn_model['enc_inp'][t]: batch_input[:,t].reshape(-1,input_dim) for t in range(input_seq_len)}
	        feed_dict.update({rnn_model['target_seq'][t]: batch_output[:,t].reshape(-1,output_dim) for t in range(output_seq_len)})
	        _, loss_t = sess.run([rnn_model['train_op'], rnn_model['loss']], feed_dict)
	        print(loss_t)

	test_seq_input = true_signal(train_data_x[-15:])

	rnn_model = build_graph(feed_previous=True)

	init = tf.global_variables_initializer()

	with tf.Session() as sess:
	    sess.run(init)
	        
	    feed_dict = {rnn_model['enc_inp'][t]: test_seq_input[t].reshape(1,1) for t in range(input_seq_len)}
	    feed_dict.update({rnn_model['target_seq'][t]: np.zeros([1, output_dim]) for t in range(output_seq_len)})
	    final_preds = sess.run(rnn_model['reshaped_outputs'], feed_dict)
	    
	    final_preds = np.concatenate(final_preds, axis = 1)        


	l1, = plt.plot(range(85), true_signal(train_data_x[:85]), label = 'Training truth')
	l2, = plt.plot(range(85, 105), y[85:], 'yo', label = 'Test truth')
	l3, = plt.plot(range(85, 105), final_preds.reshape(-1), 'ro', label = 'Test predictions')
	plt.legend(handles = [l1, l2, l3], loc = 'lower left')
	plt.show()

**Comment by aaxwaz** [comment]
I guess you shouldn't remove those lines. 
If you remove those lines and run `
rnn_model = build_graph(feed_previous=True)`, it will reset the graph and rebuild it from scratch with randomly initialized model parameters. So your test preds will be ramdom.

**Comment by zdarktknight** [comment]
Thank you for your explanation. I add those lines back and it works.
